rpcs3/Emu/RSX/Common/TextureUtils.cpp

```diff
@@ -53,6 +53,38 @@ struct copy_unmodified_block_swizzled
 	}
 };
 
+struct copy_unmodified_block_vtc
+{
+	template<typename T, typename U>
+	static void copy_mipmap_level(gsl::span<T> dst, gsl::span<const U> src, u16 width_in_block, u16 row_count, u16 depth, u32 dst_pitch_in_block, u32 src_pitch_in_block)
+	{
+
+		static_assert(sizeof(T) == sizeof(U), "Type size doesn't match.");
+		u32 row_element_count = width_in_block * row_count;
+		u32 dst_offset = 0;
+		u32 src_offset = 0;
+		const u32 vtc_tile_row_count = 4;
+
+		// Undo Nvidia VTC tiling - place each 2D texture slice back to back in linear memory
+		for (int d = 0; d < depth; d++) {
+
+			// copy one slice of the 3d texture
+			for (u32 i = 0; i < row_element_count; i += 1) {
+				copy(dst.subspan(dst_offset + i, 1), src.subspan(src_offset + i * vtc_tile_row_count, 1));
```

Comment:
Avoid byte sized subspans. This is very slow. Just grab the whole slice at once if possible and assert on size of the src span before starting the transfer.

Reply:
I initially thought the count was in bytes too.  But it is in subspans.  It will copy 8  bytes at a time for DXT1 or 16 bytes at a time for DXT5.  I verified it by stepping through the assembly code.  It uses a single MMX register to copy 16 bytes at a time.